zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,675 @@
+import logging
+import os
+import subprocess
+from typing import Any, Dict, List, Set
+
+import bson
+from django.conf import settings
+from django.forms.models import model_to_dict
+from django.utils.timezone import now as timezone_now
+
+from zerver.data_import.import_util import (
+    SubscriberHandler,
+    ZerverFieldsT,
+    build_message,
+    build_personal_subscriptions,
+    build_realm,
+    build_recipients,
+    build_stream,
+    build_stream_subscriptions,
+    build_user_profile,
+    build_zerver_realm,
+    create_converted_data_files,
+    make_subscriber_map,
+    make_user_messages,
+)
+from zerver.data_import.sequencer import NEXT_ID, IdMapper
+from zerver.data_import.user_handler import UserHandler
+from zerver.lib.emoji import name_to_codepoint
+from zerver.lib.utils import process_list_in_batches
+from zerver.models import Reaction, Recipient, UserProfile
+
+
+def make_realm(realm_id: int, realm_subdomain: str, domain_name: str) -> ZerverFieldsT:
+    NOW = float(timezone_now().timestamp())
+
+    zerver_realm = build_zerver_realm(realm_id, realm_subdomain, NOW, "RocketChat")
+    realm = build_realm(zerver_realm, realm_id, domain_name)
+
+    # We may override these later.
+    realm["zerver_defaultstream"] = []
+
+    return realm
+
+
+def process_users(
+    user_id_to_user_map: Dict[str, Dict[str, Any]],
+    realm_id: int,
+    domain_name: str,
+    user_handler: UserHandler,
+    user_id_mapper: IdMapper,
+) -> None:
+    for rc_user_id in user_id_to_user_map:
+        user_dict = user_id_to_user_map[rc_user_id]
+        user_dict["is_mirror_dummy"] = False
+
+        if user_dict["type"] != "user":
```

Comment:
Can you add a comment documenting the other types that might exist in Rocket.Chat?

Reply:
Added the comment.